src/networkextension.cs

```diff
@@ -146,10 +161,26 @@ interface NEAppProxyFlow {
 	
 		[Export ("closeWriteWithError:")]
 		void CloseWrite ([NullAllowed] NSError error);
+
+		[NoWatch, NoTV, NoiOS, Mac (10,15,4)]
+		[Export ("setMetadata:")]
+		void SetMetadata (OS_nw_parameters parameters);
 	
 		[Export ("metaData")]
 		NEFlowMetaData MetaData { get; }
 
+		[NoWatch, NoTV, Mac (10,15,4), iOS (13,4)]
+		[NullAllowed, Export ("networkInterface", ArgumentSemantic.Copy)]
+		OS_nw_interface WeakNetworkInterface { get; set; }
```

Comment:
It might be lack of caffeine (I'm working on that right now) but I don't recall using the `Weak` pattern for an `IntPtr`, which is what `OS_nw_interface` ends up to be, inside `INativeObject`.

Reply:
This is because I am providing a more concrete property that Wraps the IntPtr version, maybe I should make `WeakNetworkInterface` internal or teach the generator about [NativeObject](https://github.com/xamarin/xamarin-macios/blob/master/src/CoreFoundation/NativeObject.cs) objects.
```csharp
		[NoWatch, NoTV, Mac (10,15,4), iOS (13,4)]
		[NullAllowed, Export ("networkInterface", ArgumentSemantic.Copy)]
		OS_nw_interface WeakNetworkInterface { get; set; }

		[NoWatch, NoTV, Mac (10,15,4), iOS (13,4)]
		NWInterface NetworkInterface {
			[Wrap ("Runtime.GetINativeObject<NWInterface> (WeakNetworkInterface, false)")]
			get;
			[Wrap ("WeakNetworkInterface = value != null ? value.Handle : IntPtr.Zero")]
			set;
		}
```

Reply:
@spouliot  Thinking about it, maybe I should do both, make the property internal and in a separated PR teach the generator about [NativeObject](https://github.com/xamarin/xamarin-macios/blob/master/src/CoreFoundation/NativeObject.cs) objects.

Reply:
I'm not sure we need (at least _right now_) special generator support...

We normally use `Weak` when there are other _easy_ ways to use the API, e.g. `NSDictionary` (instead of strong dictionary) and `NSObject` for delegates (since you implement the members on a different type).

I'm not sure those applies to `IntPtr` and, as usual, it's much easier to expose it later than to remove it :)

Reply:
So making the weak internal and just exposing the wrapper it is